lib/utils/docker.coffee

```diff
@@ -266,3 +313,73 @@ exports.getDocker = (options) ->
 exports.getImageSize = (docker, image) ->
 	docker.getImage(image).inspectAsync()
 	.get('Size')
+
+hasQemu = ->
+	fs = require('mz/fs')
+
+	getQemuPath()
+	.then(fs.stat)
+	.return(true)
+	.catchReturn(false)
+
+getQemuPath = ->
+	resin = require('resin-sdk-preconfigured')
+	path = require('path')
+	fs = require('mz/fs')
+
+	resin.settings.get('binDirectory')
+	.then (binDir) ->
+		# The directory might not be created already,
+		# if not, create it
+		fs.access(binDir)
+		.catch code: 'ENOENT', ->
+			fs.mkdir(binDir)
+		.then ->
+			path.join(binDir, 'qemu-execve')
+
+platformNeedsQemu = ->
+	os = require('os')
+	os.platform() == 'linux'
+
+installQemu = ->
+	request = require('request')
+	fs = require('fs')
+	zlib = require('zlib')
+
+	qemuVersion = 'v2.5.50-resin-execve'
```

Comment:
This would be nice as a constant at the top of the module, so it's easy to see/change it